baseplate/integration/pyramid.py

```diff
@@ -71,26 +91,23 @@ def _on_new_request(self, event):
             except (KeyError, ValueError):
                 pass
 
-        request.start_root_span(request.matched_route.name, trace_info)
+        request.start_server_span(request.matched_route.name, trace_info)
+        request.trace.set_tag("http.url", request.url)
+        request.trace.set_tag("http.method", request.method)
+        request.trace.set_tag("peer.ipv4", request.remote_addr)
 
-    def _start_root_span(self, request, name, trace_info=None):
-        request.trace = self.baseplate.make_root_span(
+    def _start_server_span(self, request, name, trace_info=None):
+        request.trace = self.baseplate.make_server_span(
             request,
             name=name,
             trace_info=trace_info,
         )
         request.trace.start()
 
-    # pylint: disable=no-self-use
-    def _on_new_response(self, event):
-        if not event.request.matched_route:
-            return
-
-        event.request.trace.stop()
-
     def includeme(self, config):
-        config.add_subscriber(self._on_new_request, ContextFound)
-        config.add_subscriber(self._on_new_response, NewResponse)
+        config.add_subscriber(self._on_new_request, pyramid.events.ContextFound)
+        config.add_tween("baseplate.integration.pyramid._make_baseplate_tween",
+                         under=pyramid.tweens.EXCVIEW)
```

Comment:
Ah ok, exception views. I see. Yeah, we just want to silently observe the exception not claim any ability to render a useful response etc. So this lets us inject ourselves between the app and the pyramid layer that handles exceptions and then just watch them fly by without anyone knowing better.